### ISSUE TITLE
slow vispy.scene.SceneCanvas updates

### ISSUE THREAD

**Issue by dafred94** [vispy/vispy#1700]
I made a simple animation to illustrate a dynamic system in vispy. Using vispy.app.Timer, I update a vispy.scene.SceneCanvas every few milliseconds (goal: ~16.67ms). The "update" function itself updating attributes of objects in the scene seems to only take under 3ms, so I'm wondering why the animation becomes so slow frameratewise.. Am I making errors here? How can I speed up my animation?
I am attaching my code. To try it, run "python3 sliding_mass.py" in the uncompressed folder.
(Note that "solutions.npy" will be written to that folder to keep animation somewhat independent of simulation).
[MbO.zip](https://github.com/vispy/vispy/files/3558501/MbO.zip)

**Comment by djhoese** [comment]
I didn't run the code, but what is the shape of `solution` that gets used in `visualize_solution`?

The main problem with your current approach is the number of separate visuals you are working with, this is a common pitfall and I'll try to explain what's going on.

Each Visual object in vispy is a separate OpenGL program. Your GPU, or rather OpenGL, will only draw one program at a time. Sending the programs to the GPU isn't necessarily slow, but it takes time for the GPU to draw them. Depending on what is being drawn in the visuals this slow down can be seen with even 10 visuals. A better approach is to draw all similar shapes (your ellipses) in one OpenGL Program (vispy Visual). As a proof of concept you could try the Markers visual and see if that performs better.

The long term solution to this for more complex shapes is the "collection" objects, but they have not been updated to work with the Scene object due to the time investment that may be involved.

**Comment by dafred94** [comment]
Thanks for your immediate help!
Yes, i was basically trying to run 20 visuals at once. Your tip using a single MarkerVisual helped a lot and now the framerate is top notch.
... However, another problem became apparent when I ran the animation on a 144 Hz display: The animation is sped up a lot compared to a 60 Hz display, ALTHOUGH I manually set the interval option of the corresponding vispy.app.Timer to 1/60. Could it be that the parameter is just ignored and the refresh interval will be set just based on the display capabilities?

**Comment by djhoese** [comment]
Without re-reviewing the code (on a different computer than originally), could it be that you are accidentally sending an interval of `0` which may be treated as "run as fast as possible"?

The Timer speed, when set, should not be affected by the refresh rate of the display unless I'm missing something.

**Comment by aviogit** [comment]
I believe that interval is actually ignored for values < 0.016 seconds (i.e. 60 Hz). I'm trying to make the rendering faster, but all my attempts get ignored.

I've also tried to set `vsync=False` in `scene.SceneCanvas()` constructor but it makes no difference (I'm printing timestamps in `update()` and they are all very close to 0.016).

I think there is something OpenGL related going on here, also `glxgears` renders at 60 fps, it may be a "general setting".

**Comment by djhoese** [comment]
The `app.Timer` should be a wrapper around a backend-specific timer. So for Qt4/5-based backends you get a QTimer object. If you are concerned about your specific application/code then please file a separate issue with a minimum code example demonstrating the performance issues you are seeing.

**Comment by ariel-avi** [comment]
I just found the cause to my problem in this topic. I'm trying to create a graphical display of a mechanical system (suspension system). It basically contains cylinders and spheres, but it ran so slowly that I came to think it was a problem of too much data points (wasn't even 1M). Then I tried to use line plots (LinePlot3D tutorial) and the problem persists. Is there any way I can do it, in a way that every part works independently? A picture of what I have now follows:

![image](https://user-images.githubusercontent.com/59147674/75079379-46c95680-54e7-11ea-891c-72aa91d40405.png)

I fit into the 'scientists with no prior knowledge of OpenGL' group, btw.
